fixed padding left so the padding on sitemap and join-us page cards a…

macho-catt opened the pull request:
Fixes #1887 

### The Issue:

Basically, the page card for site map has a media query set up for screens of tablets and above to have padding-left: 0. Without this media query, The padding is off for the icon for tablets. Initially, I made the change so that the padding-left: 0 only applies when the screen is between a tablet and a desktop. This is the easier fix. 

However, when I looked at both pages, the images seemed a little off and not in the center (this is the current state of the pages), so I proceeded with solution 2: set padding-left: 0. This ensures that the images on both pages also look centered. However, this broke the join-us-donation part of the join page. To fix that, I had to add a margin-left: 48px for the tablet media query, essentially cancelling the removal of the padding for that section.

Of course solution 2 changes more than solution 1, but it looks more centered. If solution 1 is preferred, please let me know.

### What changes did you make and why did you make them ?

  - For _join-us.scss, added padding-left: 0 for page-card--join's media query so that tablet screens and wider will make the image look centered. This becomes consistent with sitemap's page card as well.
  - For _join-us.scss, added margin-left: 48px for join-us-donation media query so that this particular section still has the spacing on the left, which was removed with the above change
  - For _join-us.scss, changed the .page-card--join padding to 48px to match the padding for .page-card--sitemap in _sitemap.scss. This makes the page card's padding consistent for both pages
  - For _sitemap.scss, removed padding: 48px for page-card--sitemap-flex because this is redundant to page-card--sitemap having a padding of 48px as well
  - For _sitemap.scss, removed margin-left: auto for the img tags under page-card--sitemap. This is redundant because the same tag is set with margin: auto already. I realize that instead of removing, I commented it out, so I fixed this in a second commit.
  - For pages/sitemap.html, removed the classes page-card, card-body, and card-width from the two page card sections because page-card is redundant (declared twice) and the other two classes are not being used in any .scss file.

### Screenshots of Proposed Changes Of The Website  (if any, please do not screen shot code changes)
<!-- Note, if your images are too big, use the <img src="" width="" length="" />  syntax instead of ![image](link) to format the images -->
<!-- If images are not loading properly, you might need to double check the syntax or add a newline after the closing </summary> tag -->

<details>
<summary>Sitemap page before changes are applied</summary>

![Sitemap full page] ![sitemap before](https://user-images.githubusercontent.com/48004150/125149333-6f0e2100-e0ed-11eb-9e9e-d0ca0ab4601b.png)


</details>

<details>
<summary>Sitemap page after changes are applied</summary>
  
![Sitemap full page] ![sitemap after](https://user-images.githubusercontent.com/48004150/125149390-ad0b4500-e0ed-11eb-926f-a0df0172ee2f.png)


</details>

<details>
<summary>Join-us page before changes are applied</summary>

![Join-us full page] ![join-us before](https://user-images.githubusercontent.com/48004150/125149443-d5933f00-e0ed-11eb-8b47-a3366d7b339e.png)

![Join card desktop version] ![join card desktop before](https://user-images.githubusercontent.com/48004150/125149500-3e7ab700-e0ee-11eb-8da8-953371d02eaf.PNG)

![Join card tablet version] ![join card tablet before](https://user-images.githubusercontent.com/48004150/125149507-489cb580-e0ee-11eb-8765-356310819707.PNG)



</details>

<details>
<summary>Join-us page after changes are applied</summary>
  
![Join-us full page] ![join-us after](https://user-images.githubusercontent.com/48004150/125149448-d9bf5c80-e0ed-11eb-9fd6-7d9a2ce51b96.png)

![Join card desktop version] ![join card desktop after](https://user-images.githubusercontent.com/48004150/125149493-3589e580-e0ee-11eb-9201-fcb8f290700c.PNG)

![Join card tablet version] ![join card tablet after](https://user-images.githubusercontent.com/48004150/125149516-55b9a480-e0ee-11eb-900f-2b8e607238e5.PNG)

![Donation card broken] 
![lights on after change no inspect](https://user-images.githubusercontent.com/48004150/125149544-75e96380-e0ee-11eb-984e-d803ebf89fea.PNG)



</details>

</details>

<details>
<summary>Join-us page after the fix for donation card applied</summary>
  
![Join-us full page] ![join-us after fix left margin](https://user-images.githubusercontent.com/48004150/125149465-f8255800-e0ed-11eb-89c7-a6b6e0d66f6d.png)

![Donation card fixed] 
![lights on after change fixed left margin no inspect](https://user-images.githubusercontent.com/48004150/125149569-8dc0e780-e0ee-11eb-9fdb-8799977bc411.PNG)



</details>

macho-catt committed:
fixed padding left so the padding on sitemap and join-us page cards are consistent

macho-catt committed:
removed comment

macho-catt committed:
Merge branch 'fix-large-icon-page-cards-inconsistency' of https://github.com/macho-catt/website into fix-large-icon-page-cards-inconsistency

erikaBell reviewed (commented):
I noticed that each page also has quite a difference in margin between the large icon and the right handed text, causing the text position to sit differently on each page. 

<details>
<summary>join-us</summary>

<img width="600" alt="Screen Shot 2021-07-12 at 10 32 09 PM" src="https://user-images.githubusercontent.com/67438372/125396611-a08f2280-e361-11eb-9298-509042b045c3.png">

</details>

<details>
<summary>site map</summary>

<img width="543" alt="Screen Shot 2021-07-12 at 10 31 57 PM" src="https://user-images.githubusercontent.com/67438372/125396638-a97ff400-e361-11eb-88f4-5f523f1333a3.png">

</details>

Would it be possible for @macho-catt to also add this into the pr? @daniellex0 / @averdin2 / @Aveline-art, or is a new issue preferred?

Aveline-art commented:
@erikaBell It might be easy enough to tack on these additional changes and leave a follow-up comment, but I will let @macho-catt decide on whether to make the changes or make a new issue.

macho-catt commented:
I could definitely do that. However, it seems like it would be better to make the page cards on the two pages (join and sitemap) to be exactly the same. Right now the two have different classes that delegate their style (page-card--join for join page and page-card--sitemap-flex + page-card--sitemap for sitemap page). 
Instead of making changes on those classes to make the page cards the same, should I just create a generic css class that designs page cards with large icons, and use that for those two? That will also make it re-usable for the future. And if that's the case, should I put this template under _cards.scss, or is there another place? @Aveline-art @daniellex0 @averdin2

Edit: I just realized that the page card on join page is using grid, while the one on sitemap is using flex. This may also attribute to why there are certain differences in how they look. If we want to make the two the same, then we have to pick either grid or flex and use it for both, and potentially refactor / rewrite the code so both pages use the same classes.

jbubar commented:
I want to add my two cents. I think that it should have the padding-left (option 1), and the stuff on the inside should be newly styled to fit and look centered. 

I think it makes more sense if our page-cards have a uniform padding around the whole card. We can talk about this in the dev meeting?

Aveline-art commented:
> should I just create a generic css class that designs page cards with large icons, and use that for those two? 

Designing a page card w/ icons component might render it less reusable, ironically, since that component could only be used when 1) a card is needed, and 2) an icon is needed. We should discuss this during the meeting, but I will forward that we add the icon styling into the elements/_icon.scss folder. This way, when designing a page with icons inside cards, the classes can be pulled separately and mixed and matched.

As for grid vs flex, that is over my head! Maybe @jbubar or @abuna1985 can give their 2 cents?

Good question, @macho-catt.

daniellex0 commented:
Btw just to add a note about "page-card--sitemap" etc. classes- that is the standard way to add a page-specific class to a reusable component like a page card. But you're right, it could be good to have a reusable class for these icon page cards so that they are... more reusable! Maybe "page-card--large-icon" or something like that (sorry I'm really bad at naming, but you get the gist), so that it's still clear it's connected to the page card standardized classes?

macho-catt committed:
Merge branch 'gh-pages' into fix-large-icon-page-cards-inconsistency

macho-catt committed:
refactor-css-code-for-new-page-card-template

macho-catt commented:
Added a new commit: 9cbde10
### Changes:

- _cards.scss: created new class: page-card--large-icon-container, which acts as the template container for page cards with large icons. This class has another class inside it, page-card--large-icon-body, to style the body inside the container. The two classes have media queries for mobile, tablet, and desktop
- _join-us.scss: removed attributes that now go to the new template class
- _sitemap.scss: removed attributes that now go to the new template class.
- join-us.html: added new class to page card and div in the page card
- sitemap.html: previously the page card had two img tag, with one image hidden at a time depending on the screen. I updated this so the page only calls one img, and changes how it's displayed depending on the scene. This also involved adding the new class to the page card and div in the page card

I can definitely add more adjustments if needed.

### Screenshots
<details>
<summary>Sitemap desktop full page before</summary>

![sitemap-original](https://user-images.githubusercontent.com/48004150/125735012-91ea9a30-8905-4014-a64d-606a1c7d5f91.png)
</details>

<details>
<summary>Sitemap desktop full page after</summary>

![sitemap-new](https://user-images.githubusercontent.com/48004150/125735085-88a92661-7ca9-4449-8cbe-2064a47cfb01.png)
</details>

<details>
<summary>Sitemap tablet before</summary>

![sitemap-tablet-original](https://user-images.githubusercontent.com/48004150/125735115-199bb089-2527-4f92-9efe-3456eeafde3c.png)
</details>

<details>
<summary>Sitemap tablet after</summary>

![sitemap-tablet-new](https://user-images.githubusercontent.com/48004150/125735126-425f5a88-bcb5-4cfc-8ab9-bf6db394b2a5.png)
</details>

<details>
<summary>Sitemap mobile before</summary>

![sitemap-mobile-original](https://user-images.githubusercontent.com/48004150/125735174-ced11777-f420-4f1c-99a2-b3a9d7db27cd.png)
</details>

<details>
<summary>Sitemap mobile after</summary>

![sitemap-mobile-new](https://user-images.githubusercontent.com/48004150/125735190-b8c5fe81-f50a-4e05-b188-de9a1f6e0001.png)
</details>

<details>
<summary>Join-us desktop full page before</summary>

![join-original](https://user-images.githubusercontent.com/48004150/125735208-3e4cb7ac-d705-4550-bbcf-ea08b3845cab.png)
</details>

<details>
<summary>Join-us desktop full page after</summary>

![join-new](https://user-images.githubusercontent.com/48004150/125735228-66d3d481-aef3-4b6a-99c3-5c8f28424bb8.png)
</details>

<details>
<summary>Join-us tablet before</summary>

![join-tablet-original](https://user-images.githubusercontent.com/48004150/125735250-0ba8b2f7-3738-4c19-930e-08a27a8e50cc.png)
</details>

<details>
<summary>Join-us tablet after</summary>

![join-tablet-new](https://user-images.githubusercontent.com/48004150/125735260-6da6bfd4-a4a2-44f9-8c0d-f5acaed542a5.png)
</details>

<details>
<summary>Join-us mobile before</summary>

![join-mobile-original](https://user-images.githubusercontent.com/48004150/125735275-d96776bc-fce9-4558-83a8-a797e2b2c01b.png)
</details>

<details>
<summary>Join-us mobile after</summary>

![join-mobile-new](https://user-images.githubusercontent.com/48004150/125735288-8b371558-c5b1-409c-b89a-556b25bdc109.png)
</details>

averdin2 commented:
@macho-catt As per your comment https://github.com/hackforla/website/pull/1918#issuecomment-881036944 on PR #1918, I merged that PR. I can help you with any merge conflicts you are facing after that merge.

averdin2 commented on pages/sitemap.html:
```diff
@@ -25,51 +25,48 @@ <h1>Site Map</h1>
+      </div>
     </div>
 
 
-    <div class="page-card card-primary page-card-lg page-card--sitemap page-card--sitemap-flex page-card card-body card-width">
-        <img class="card-desktop-icon" src="/assets/images/sitemap/site.svg">
-        <div>
-            <h3>Help Us Add Sections to the Website</h3>
-            <img class="card-mobile-icon" src="/assets/images/sitemap/site.svg">
-            <p>
-                We are looking for volunteers across all
-                disciplines (UI/UX, Development, Data Science,
-                Product, etc.) to help us improve the Hack for
-                LA website! If you have ideas on how we can
-                enhance the experience and/or are passionate
-                about web development, please consider joining
-                our team.
-            </p>
-
-            <a href="https://github.com/hackforla/website/wiki" target="_blank"><button
-                    class="btn btn-primary btn-md-long-2">Check out
-                    our Wiki</button></a>
-        </div>
+    <div class="page-card card-primary page-card-lg page-card--large-icon-container">
+      <h1>Help Us Add Sections to the Website</h1>  
```
@macho-catt I pulled your PR with the merged changes from PR #1918. The main conflict I am seeing is that you need to add (class="title4") on lines 33 and 54. Also, what is your reasoning for changing the h3 tag to an h1 on lines 33 and 54? I actually think they should be an h2 (also on line 25).

macho-catt replied:
@averdin2 Since the ask was to standardize the page cards with large icons, I based the template off of join-us.html, which used the h1 tags, and created the template (cards.scss) using those tags. I also thought of the other way around (changing tags in join-us to h3) since it made more sense, but I went in this direction instead to simplify refactoring the css from join-us.scss to cards.scss. I could certainly change the tags for the sitemap page to the most reasonable one and add the proper class, but I would also need to change the tags for join-us (and revise the new class template I created) to match the changes. Not a big deal at all. Other than the merge conflicts, are there other things that you see that could be improved?

Thanks!

averdin2 replied:
@macho-catt This is actually an interesting discussion. I want to see what the rest of the team thinks about the header tags. In terms of other changes, I saw that there were other reviewers on this issue already so I am not doing a formal review. I only really looked at the sitemap page, but it looks good to me 👍

macho-catt committed:
fixed merge conflicts and added responsiveness for headers

macho-catt commented:
### Changes
 - as per @averdin2, I added the class "title4" to take into account a conflicting merge from a different PR
 - Changed h1 and h2 tags under cards.scss to classes to make them more usable 
 - Made the class for h2 to be responsive when the screen is below a tablet

<details>
<summary>Sitemap full page after fixing merge conflict</summary>

![sitemap-full-after-merge](https://user-images.githubusercontent.com/48004150/125885526-235d7058-c7a0-44c3-8668-b434ee8afe88.png)

</details>

<details>
<summary>Sitemap tablet view after fixing merge conflict</summary>

![sitemap-tablet-after-merge](https://user-images.githubusercontent.com/48004150/125885534-b03fb283-e4ea-417b-991b-cc0f4c19588f.png)

</details>

<details>
<summary>Sitemap mobile view after fixing merge conflict</summary>

![sitemap-mobile-after-merge](https://user-images.githubusercontent.com/48004150/125885561-50c2b16f-4a3b-4ca6-a005-e5c52632a877.png)

</details>

<details>
<summary>Join full page after fixing merge conflict</summary>

![join-full-after-merge](https://user-images.githubusercontent.com/48004150/125885596-8f5982a8-64e2-4724-81e0-1714619aff92.png)


</details>

<details>
<summary>Join tablet view after fixing merge conflict</summary>

![join-tablet-after-merge](https://user-images.githubusercontent.com/48004150/125885605-a4a42b0b-9c6c-4981-8b6f-d4dbe9bf81b4.png)


</details>

<details>
<summary>Join mobile view after fixing merge conflict</summary>

![join-mobile-after-merge](https://user-images.githubusercontent.com/48004150/125885613-b74509c6-8cab-42e2-a317-a928f9122f4f.png)


</details>

macho-catt committed:
accept changes to resolve merge conflicts

macho-catt committed:
changes to adapt new template

macho-catt commented:
New commits: f7f6d1b, 0f9f3d0, ac76ff5

f7f6d1b was the initial commit but the merge conflict was not resolved, so I resolved it first in 0f9f3d0, then reapplied the changes to adapt the new template in ac76ff5

### Changes
 - as per @averdin2, to resolve a conflicting merge from a different PR, I accepted the incoming change from the other PR first. I then went ahead and removed one of the two img tags in the section since the new css template fixes the responsiveness (instead of hiding the img tag for the responding display screen) as well as remove the classes that became unused as a result of the css refactoring. Finally, I added the new classes to apply the template.
 - Changed h1 and h2 tags under cards.scss to classes to make them more usable 
 - Made the class for h2 to be responsive when the screen is below a tablet

<details>
<summary>Sitemap full page after fixing merge conflict</summary>

![sitemap-full-after-merge](https://user-images.githubusercontent.com/48004150/125885526-235d7058-c7a0-44c3-8668-b434ee8afe88.png)

</details>

<details>
<summary>Sitemap tablet view after fixing merge conflict</summary>

![sitemap-tablet-after-merge](https://user-images.githubusercontent.com/48004150/125885534-b03fb283-e4ea-417b-991b-cc0f4c19588f.png)

</details>

<details>
<summary>Sitemap mobile view after fixing merge conflict</summary>

![sitemap-mobile-after-merge](https://user-images.githubusercontent.com/48004150/125885561-50c2b16f-4a3b-4ca6-a005-e5c52632a877.png)

</details>

<details>
<summary>Join full page after fixing merge conflict</summary>

![join-full-after-merge](https://user-images.githubusercontent.com/48004150/125885596-8f5982a8-64e2-4724-81e0-1714619aff92.png)


</details>

<details>
<summary>Join tablet view after fixing merge conflict</summary>

![join-tablet-after-merge](https://user-images.githubusercontent.com/48004150/125885605-a4a42b0b-9c6c-4981-8b6f-d4dbe9bf81b4.png)


</details>

<details>
<summary>Join mobile view after fixing merge conflict</summary>

![join-mobile-after-merge](https://user-images.githubusercontent.com/48004150/125885613-b74509c6-8cab-42e2-a317-a928f9122f4f.png)


</details>

adam-abundis commented:
@daniellex0 I was reviewing the pull request and referenced the Figma file for `Join Us` and `Sitemap` page and noticed there is no tablet mockup.

<details>
  <summary>Here is what the tablet view of what the cards would look like on the Sitemap page</summary>
  <img src="https://user-images.githubusercontent.com/48004150/125735250-0ba8b2f7-3738-4c19-930e-08a27a8e50cc.png" />
</details>

<details>
  <summary>Here is what the tablet view of what the cards would look like on the Join Us page</summary>
  <img src="https://user-images.githubusercontent.com/48004150/125735260-6da6bfd4-a4a2-44f9-8c0d-f5acaed542a5.png" />
</details>

There is a lot of white space within the card, but it matches the margins within mobile view . Would this be preferred or should we add more margin around the card (Add gray space around the card) to match the desktop view?

I am thinking we stay consistent and add margin around the card (add gray space) so the card is consistent in all 3 views.

abuna1985 commented:
@daniellex0 @macho-catt I might just convert my question/query into a separate issue since this pull request has been on here for a week. I will continue the review for now.

macho-catt committed:
Merge branch 'gh-pages' into fix-large-icon-page-cards-inconsistency

averdin2 commented:
@macho-catt As we discussed on Sunday's meeting, in the sitemap.html and join-us.html, you can make all the other h1 tags that are not the main header tag into h2 tags.

macho-catt committed:
Merge branch 'gh-pages' into fix-large-icon-page-cards-inconsistency

macho-catt committed:
changed header tags to appropriate ones and used title classes to define font size

macho-catt commented:
New commit: 5386e81 

### Changes
- Updated header tags as advised by @averdin2 
- Updated each headers in join-us.html to use the proper title classes to get font sizes.
- Removed font sizes from cards.scss so the headers take the font size from the title classes

<details>
<summary>Sitemap full page</summary>

![sitemap-full-desktop](https://user-images.githubusercontent.com/48004150/126234453-899ef596-9266-4c64-af87-64a13b82ed7d.png)


</details>

<details>
<summary>Sitemap tablet view</summary>

![sitemap-tablet](https://user-images.githubusercontent.com/48004150/126234474-d79a32a7-5b33-4159-b1ba-6978f4fd8532.png)


</details>

<details>
<summary>Sitemap mobile view</summary>

![sitemap-mobile](https://user-images.githubusercontent.com/48004150/126234486-35a653a1-2671-431a-94cd-c8407ceeeaab.png)


</details>

<details>
<summary>Join full page</summary>

![join-full-desktop](https://user-images.githubusercontent.com/48004150/126234503-c615a4db-2227-4c62-8580-add0c6753c73.png)


</details>

<details>
<summary>Join tablet view</summary>

![join-tablet](https://user-images.githubusercontent.com/48004150/126234519-ef2a0eca-ace7-4f10-8286-af4442ebb6cf.png)


</details>

<details>
<summary>Join mobile view</summary>

![join-mobile](https://user-images.githubusercontent.com/48004150/126234533-752446bc-2e51-463c-9756-5e599bd6f2bb.png)



</details>

@daniellex0 @abuna1985 please let me know if you want me to fix the margins for the tablet view on the page cards. Thanks!

daniellex0 commented:
This all looks good to me! (just tagging @kristine-eudey to see PR reviews)

erikaBell reviewed (approved):
Great work @macho-catt.

Lgtm ~*
